src/python/zquantum/core/circuit/conversions/pyquil_conversions.py

```diff
@@ -203,3 +221,51 @@ def convert_circuit_to_pyquil(
         program += convert_to_pyquil(gate, program)
 
     return program
+
+
+@singledispatch
+def convert_from_pyquil(obj: Union[pyquil.Program, pyquil.quil.Gate]):
+    pass
```

Comment:
I know this isn't finished yet, but would it be better here to raise an `Exception` by default?

Reply:
Great idea! Fixed in 2d8f706